master/buildbot/db/changes.py

```diff
@@ -195,7 +195,8 @@ def getChangesForBuild(self, buildid):
 
         # For each codebase, append changes until we match the parent
         for cb, change in iteritems(fromChanges):
-            if change and change['changeid'] != toChanges.get(cb, {}).get('changeid'):
+            # Careful; toChanges[cb] may be None from getChangeFromSSid
+            if change and change['changeid'] != (toChanges.get(cb) or {}).get('changeid'):
```

Comment:
Yes, `get(cb, {})` is the original line; it doesn't work because of this:

```
toChanges[cb] = None
badValue = toChanges.get(cb, {})  # badValue = None 
badValue.get('changeId')  # e.g. None.get(), which throws an exception
```

I'm not sure what @tardyp's comment was meant to suggest, but the code as is doesn't reliably work unless changes are made elsewhere to guarantee a dictionary rather than None is returned.